drivers/pwm/pwm_stm32.c

```diff
@@ -195,62 +195,37 @@ static int pwm_stm32_init(struct device *dev)
 	return 0;
 }
 
+#define PWM_DEVICE_INIT_STM32(n, apb, prescaler)			  \
+	static struct pwm_stm32_data pwm_stm32_dev_data_ ## n = {	  \
+		/* Default case */					  \
+		.pwm_prescaler = prescaler,				  \
+	};								  \
+									  \
+	static const struct pwm_stm32_config pwm_stm32_dev_cfg_ ## n = {  \
+		.pwm_base = TIM ## n ## _BASE,				  \
+		.pclken = { .bus = STM32_CLOCK_BUS_ ## apb,		  \
+			    .enr = LL_ ## apb ## _GRP1_PERIPH_TIM ## n }, \
+	};								  \
+									  \
+	DEVICE_AND_API_INIT(pwm_stm32_ ## n,				  \
+			    CONFIG_PWM_STM32_ ## n ## _DEV_NAME,	  \
+			    pwm_stm32_init,				  \
+			    &pwm_stm32_dev_data_ ## n,			  \
+			    &pwm_stm32_dev_cfg_ ## n,			  \
+			    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,\
+			    &pwm_stm32_drv_api_funcs);
 
 #ifdef CONFIG_PWM_STM32_1
-static struct pwm_stm32_data pwm_stm32_dev_data_1 = {
-	/* Default case */
-	.pwm_prescaler = 10000,
-};
-
-static const struct pwm_stm32_config pwm_stm32_dev_cfg_1 = {
-	.pwm_base = TIM1_BASE,
-	.pclken = { .bus = STM32_CLOCK_BUS_APB2,
-		    .enr = LL_APB2_GRP1_PERIPH_TIM1 },
-};
-
-DEVICE_AND_API_INIT(pwm_stm32_1, CONFIG_PWM_STM32_1_DEV_NAME,
-		    pwm_stm32_init,
-		    &pwm_stm32_dev_data_1, &pwm_stm32_dev_cfg_1,
-		    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
-		    &pwm_stm32_drv_api_funcs);
+/* 16-bit advanced-control timer */
+PWM_DEVICE_INIT_STM32(1, APB2, 10000)
```

Comment:
Is that possible to keep "TIM_" prefix in the timer argument?
"1" is not quite meaningful."TIM_1" would help reader to know what this is about without need to go back to the macro definition.

Reply:
Ok this looks reasonable. I will do it.

Reply:
Mmmh but there is an issue with your proposal. If I replace "1" by "TIM1", this will be used at other places in the maro, like naming the cfg and data structures (that's ok I guess) and the DEV_NAME Kconfig. Thus CONFIG_PWM_STM32_1_DEV_NAME will be renamed to CONFIG_PWM_STM32_TIM1_DEV_NAME. I am not sure if this is desirable.

A workaround would be to have 2 parameters to the macro (like `PWM_DEVICE_INIT_STM32(TIM1, 1, ...)`) but this is not very effective.

Reply:
Ok, nevermind then, thanks for having had a look.